### ISSUE TITLE
Authenticate with Oauth

### ISSUE THREAD

**Issue by ruben-janssens** [grafana/terraform-provider-grafana#1964]
### Feature Request

Our Grafana is set up with authentication via oauth and we want to eliminate usernames/passwords completely.
Would it somehow be possible to add oauth authentication to the provider.

**Comment by Duologic** [comment]
Do you want to authenticate the `provider {}` block with this? You can already use API tokens for it.

**Comment by ruben-janssens** [comment]
Yes we want to authenticate the provider with oauth (or oidc). Something in the likes of AWS or Azure

**Comment by jefferson-lima** [comment]
I have the same issue. Using API tokens would require a manual step of creating the tokens, I would like to have a fully automated setup where I deploy Grafana and then use the provider to configure datasources, dashboards, etc.

**Comment by Duologic** [comment]
This would be a feature request for the Grafana API ([supports basic auth or tokens](https://grafana.com/docs/grafana/latest/developer-resources/api-reference/http-api/authentication/)) or the Grafana Cloud API ([supports tokens](https://grafana.com/docs/grafana/latest/developer-resources/api-reference/cloud-api/)). If the upstream APIs do not support oauth or oidc, then the TF provider can't implement it either.
